lib/membrane/rtp/tcp_depayloader.ex

```diff
@@ -0,0 +1,119 @@
+defmodule Membrane.RTP.TCP.Depayloader do
+  @moduledoc """
+  This element provides functionality of depayloading RTP Packets received by TCP and redirecting
+  RTSP messages received in the same stream. The encapsulation is described in RFC 7826 Section 14.
+  """
+  use Membrane.Filter
+
+  alias Membrane.{Buffer, RemoteStream, RTP, RTSP}
+
+  def_options rtp_channel_id: [
+                spec: non_neg_integer(),
+                default: 0,
+                description: """
+                Channel identifier which encapsulated RTP packets will have.
+                """
+              ],
+              rtsp_session: [
+                spec: pid() | nil,
+                default: nil,
+                description: """
+                PID of a RTSP Session (returned from Membrane.RTSP.start or Membrane.RTSP.start_link)
+                that received RTSP responses will be forwarded to. If nil the responses will be
+                discarded.
+                """
+              ]
+
+  def_input_pad :input, accepted_format: %RemoteStream{type: :bytestream}
+
+  def_output_pad :output, accepted_format: %RemoteStream{type: :packetized, content_format: RTP}
+
+  @impl true
+  def handle_init(_ctx, opts) do
+    state =
+      Map.from_struct(opts)
+      |> Map.merge(%{
+        unprocessed_data: <<>>
+      })
+
+    {[], state}
+  end
+
+  @impl true
+  def handle_stream_format(:input, _stream_format, _ctx, state) do
+    stream_format = %RemoteStream{type: :packetized, content_format: RTP}
+    {[stream_format: {:output, stream_format}], state}
+  end
+
+  @impl true
+  def handle_buffer(
+        :input,
+        %Buffer{payload: payload, metadata: metadata},
+        _ctx,
+        %{unprocessed_data: unprocessed_data, rtsp_session: rtsp_session} = state
+      ) do
+    unprocessed_data =
+      if rtsp_response?(unprocessed_data, payload) do
+        if rtsp_session != nil do
+          {:ok, %RTSP.Response{status: 200}} =
+            RTSP.handle_response(rtsp_session, unprocessed_data)
+        end
+
+        <<>>
+      else
+        unprocessed_data
+      end
```

Comment:
[NIT] You can consider using `cond` instead of nested if, but do it as you like :p

Reply:
```
cond do
  not rtsp_response?(state.unprocessed_data, payload) ->
    state.unprocessed_data

  state.rtsp_session != nil ->
    {:ok, %RTSP.Response{status: 200}} =
      RTSP.handle_response(state.rtsp_session, state.unprocessed_data)
    <<>>

  state.rtsp_session == nil ->
    <<>>
end
``` 
Something like this seems reasonable, but the fact that `<<>>` is always returned if `rtsp_response?` returns true makes me prefer the nested if